backend/app/rest/api/rest_test.go

```diff
@@ -207,6 +209,61 @@ func TestRest_rejectAnonUser(t *testing.T) {
 	assert.Equal(t, http.StatusOK, resp.StatusCode, "real user")
 }
 
+func TestRest_Search(t *testing.T) {
+	var searchIndexPath string
+	ts, _, teardown := startupT(t, func(srv *Rest) {
+		var err error
+		searchIndexPath, err = randomPath(srv.WebRoot, "test-search-remark", "")
+		require.NoError(t, err)
+		srv.DataService.SearchService, err = search.NewService([]string{"remark42"}, search.ServiceParams{
+			IndexPath: searchIndexPath,
+			Analyzer:  "standard",
+		})
+		require.NoError(t, err)
+	})
+	defer teardown()
+	defer func() { _ = os.RemoveAll(searchIndexPath) }()
+
+	addComment(t, store.Comment{Text: "test test", Locator: store.Locator{SiteID: "remark42", URL: "https://radio-t.com/abc"}}, ts)
+	addComment(t, store.Comment{Text: "test hello", Locator: store.Locator{SiteID: "remark42", URL: "https://radio-t.com/blah"}}, ts)
+	addComment(t, store.Comment{Text: "hello world", Locator: store.Locator{SiteID: "remark42", URL: "https://radio-t.com/blah"}}, ts)
+	addComment(t, store.Comment{Text: "# title\n\nok\n", Locator: store.Locator{SiteID: "remark42", URL: "https://radio-t.com/blah"}}, ts)
+
+	tbl := []struct {
+		query string
+		count int
+	}{
+		{"blah", 0},
+		{"h1", 0},
+		{"world", 1},
+		{"title", 1},
+		{"\"hello world\"", 1},
+		{"test", 2},
+		{"hello", 2},
+		{"hello world", 2},
+		{"hello world test", 3},
+	}
+
+	for i, tt := range tbl {
+		tt := tt
+		t.Run(strconv.Itoa(i), func(t *testing.T) {
+			resp, err := http.Get(fmt.Sprintf("%s/api/v1/search?site=remark42&query=%s", ts.URL, url.QueryEscape(tt.query)))
+			require.NoError(t, err)
+
+			defer resp.Body.Close()
+			body, err := io.ReadAll(resp.Body)
+			require.NoError(t, err)
+			assert.Equal(t, http.StatusOK, resp.StatusCode)
+
+			result := commentsWithTotal{}
+			err = json.Unmarshal(body, &result)
+			require.NoError(t, err)
+
+			require.Len(t, result.Comments, tt.count)
+		})
+	}
+}
```

Comment:
there is no test using offset/limit, and there are no test for unhappy path. Would be good to have them all added

Reply:
How sophisticated should it be? In this test, we need to test that it works trough the query params, so a test with three comments is suitable, is it?